onnx/defs/math/defs.cc

```diff
@@ -2594,4 +2594,157 @@ ONNX_OPERATOR_SET_SCHEMA(
             propagateShapeFromInputToOutput(ctx, 0, 1);
           }
         }));
+
+static int64_t fft_signal_ndim_default = 1;
+
+static const char* FFT_ver13_doc = R"DOC(
+Fast Fourier Transform.
+
+This implementation suppose the DFT (Discrete Fourier Transform) defined by
+```
+A_k =  \sum_{m=0}^{n-1} a_m \exp\left\{-2\pi i{mk \over n}\right\}
+\qquad k = 0,\ldots,n-1.
+```
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    FFT,
+    13,
+    OpSchema()
+        .SetDoc(FFT_ver13_doc)
+        .Attr(
+            "signal_ndim",
+            "The number of dimension of the input signal."
+            "Values can be 1, 2 or 3.",
+            AttributeProto::INT,
+            fft_signal_ndim_default)
+        .Input(
+            0,
+            "input",
+            "A complex signal of dimension signal_ndim."
```

Comment:
ONNX introduced complex-numbers as a type, though no ops yet exist to support them. Seems worthwhile to make a call on this now. We could use complex as a type-specification here. If not, may be we should omit the complex-type from ONNX, if we settle on using the more basic float types.

Reply:
Supported complex would make it easier to convert models into ONNX. Otherwise, the converter needs to remember which tensor is complex. But having complex means also extend operator Cast to convert the complex tensor into real. But if FFT operator does not produce complex, removing complex everywhere would be less confusing.